hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -355,6 +356,22 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		return ctrl.Result{}, fmt.Errorf("failed to look up release image metadata: %w", err)
 	}
 
+	// Validate the NodePool and management cluster CPU architectures match if the HostedCluster does not support multi-arch NodePools
```

Comment:
This logic needs to be unit tested

Reply:
Moved this into its own function to do so. Will add a unit test before merging PR.